### DISCUSSION TITLE
Ability to add documentation to a mapping rule

### DISCUSSION THREAD

**Discussion by gordonmilligan-uod** [Health-Informatics-UoN/carrot#61] [Ideas]
This is linked to issue #628 

To improve the learnings that can be gained from mappings, it would be good if there was the option to create a short note when you generate a mapping to document the decision process that was made for that mapping rule.

**Comment by spco** [comment]
This is a really nice idea and wouldn't take too much work. Likely add a column to the ScanReportConcept table, and an input box to the FE view. Might require some thought in the case where there are multiple SRConcepts for a single Value or Field.

**Comment by AndyRae** [comment]
I would be interested in your perspective on this one @DrUrwin @armando-mv 
Given the loss of precision in mappings that we've spoke about, this functionality would feel very helpful to me (and also straightforward to implement).

**Comment by AndyRae** [comment]
I've moved this to a discussion - as it needs wider thought, design, and planning. 
I have some initial ideas around this which I will share as well.

**Comment by AndyRae** [comment]
I think a good opportunity to do this aligns with Health-Informatics-UoN/carrot#57 - with SSSOM, there is the additional `comment` field which is designed for: 
`Free text field containing either curator notes or text generated by tool providing additional informative information. `

Additionally, mappings could be enriched with further context - `confidence`, `author_id`, [`predicate_id` ](https://github.com/mapping-commons/semantic-mapping-vocabulary/blob/main/semapv-properties.tsv).

Effectively if mappings are to be understood and learnt from - the who/when/why/how should be added, right now we just have the what.
